src/main/rx/cc2500_redpine.c

```diff
@@ -0,0 +1,201 @@
+#ifdef USE_RX_REDPINE_SPI
+
+#include "build/build_config.h"
+#include "build/debug.h"
+
+#include "pg/rx.h"
+#include "pg/rx_spi.h"
+
+#include "common/maths.h"
+#include "common/utils.h"
+
+#include "config/feature.h"
+
+#include "drivers/adc.h"
+#include "drivers/rx/rx_spi.h"
+#include "drivers/rx/rx_cc2500.h"
+#include "drivers/io.h"
+#include "drivers/io_def.h"
+#include "drivers/io_types.h"
+#include "drivers/resource.h"
+#include "drivers/system.h"
+#include "drivers/time.h"
+
+#include "fc/config.h"
+
+#include "rx/rx_spi_common.h"
+#include "rx/cc2500_common.h"
+#include "rx/cc2500_frsky_common.h"
+#include "rx/cc2500_frsky_shared.h"
+
+#include "cc2500_redpine.h"
+
+extern const uint16_t crcTable[];
+
+
+#define TOTAL_PACKET_TIME 1500
+#define CHANNEL_START 3
+void redpineSetRcData(uint16_t *rcData, const uint8_t *packet)
+{
+    uint16_t channelValue;
+
+    //4 stick channels (11-bit)
+    channelValue = (uint16_t)((packet[CHANNEL_START+1] << 8) & 0x700) | packet[CHANNEL_START];
+    rcData[0] = channelValue * 2.0f / 3 + 860 - 64 * 2 / 3;
+    
+    channelValue = (uint16_t)((packet[CHANNEL_START+2] << 4) & 0x7F0) | ((packet[CHANNEL_START+1] >> 4) & 0xF);
+    rcData[1] = channelValue * 2.0f / 3 + 860 - 64 * 2 / 3;
+    
+    channelValue = (uint16_t)((packet[CHANNEL_START+4] << 8) & 0x700) | packet[CHANNEL_START+3];
+    rcData[2] = channelValue * 2.0f / 3 + 860 - 64 * 2 / 3;
+    
+    channelValue = (uint16_t)((packet[CHANNEL_START+5] << 4) & 0x7F0) | ((packet[CHANNEL_START+4] >> 4) & 0xF);
+    rcData[3] = channelValue * 2.0f / 3 + 860 - 64 * 2 / 3;    
+    //12 1-bit aux channels - first 4 are interleaved with stick channels
+    rcData[4]= (packet[CHANNEL_START + 1] & 0x08) ? PWM_RANGE_MAX : PWM_RANGE_MIN; 
+    rcData[5]= (packet[CHANNEL_START + 2] & 0x80) ? PWM_RANGE_MAX : PWM_RANGE_MIN; 
+    rcData[6]= (packet[CHANNEL_START + 4] & 0x08) ? PWM_RANGE_MAX : PWM_RANGE_MIN; 
+    rcData[7]= (packet[CHANNEL_START + 5] & 0x80) ? PWM_RANGE_MAX : PWM_RANGE_MIN; 
+    rcData[8]= (packet[CHANNEL_START + 6] & 0x01) ? PWM_RANGE_MAX : PWM_RANGE_MIN; 
+    rcData[9]= (packet[CHANNEL_START + 6] & 0x02) ? PWM_RANGE_MAX : PWM_RANGE_MIN; 
+    rcData[10]= (packet[CHANNEL_START + 6] & 0x04) ? PWM_RANGE_MAX : PWM_RANGE_MIN; 
+    rcData[11]= (packet[CHANNEL_START + 6] & 0x08) ? PWM_RANGE_MAX : PWM_RANGE_MIN; 
+    rcData[12]= (packet[CHANNEL_START + 6] & 0x10) ? PWM_RANGE_MAX : PWM_RANGE_MIN; 
+    rcData[13]= (packet[CHANNEL_START + 6] & 0x20) ? PWM_RANGE_MAX : PWM_RANGE_MIN; 
+    rcData[14]= (packet[CHANNEL_START + 6] & 0x40) ? PWM_RANGE_MAX : PWM_RANGE_MIN; 
+    rcData[15]= (packet[CHANNEL_START + 6] & 0x80) ? PWM_RANGE_MAX : PWM_RANGE_MIN;         
+}
+
+rx_spi_received_e redpineHandlePacket(uint8_t * const packet, uint8_t * const protocolState)
+{
+    static bool ledIsOn;
+    static uint16_t looptime = TOTAL_PACKET_TIME;
+    static timeUs_t packetTimerUs;
+
+    static timeUs_t totalTimerUs;
+ 
+    static uint8_t channelsToSkip = 1;
```

Comment:
This variable never changes, maybe use #define CHANNELS_TO_SKIP 1 and use it in nextChannel()

Reply:
Good catch, pulled that out.  Just changed to nextChannel(1); since that is what other protocols are doing when hopping to the next channel (with no skips)